small-talk/app/chat/page.js

```diff
@@ -1,14 +1,66 @@
 "use client"
 import Link from 'next/link'
-import React from 'react';
 import ThemeLayout from '../components/ThemeLayout';
+import socket from "../../util/socket";
+import React, { useEffect, useState } from 'react';
 
-export default function chat() {
-    return (
+export default function Chat(){
+	const [text, setText] = useState(null);
+	const [user, setUser] = useState(null);
+	  
+	useEffect(() => {
+		socket.auth = "random";
+    	socket.connect();
+		setText("whatever");
+		
+		// getting all currently connected users
+		socket.on("users", (receivedUsers) => {
+            console.log("Received users:", receivedUsers);
+            receivedUsers.forEach((user) => {
+                user.self = user.userID === socket.id;
+            });
+            setUser(receivedUsers);
+        });
+		
+		// set the event to connected users
+		socket.on("user connected", (user) => {
+			console.log("User connected:", connectedUser);
```

Comment:
I may be incorrect, but is connectedUser meant to be user? I don't see connectedUser defined.

Reply:
+1

Reply:
Thank you! I just uploaded the corrected version.